src/Spago/Config.hs

```diff
@@ -398,6 +492,27 @@ withConfigAST transform = do
         else logDebug "Transformed config is the same as the read one, not overwriting it"
       pure exprHasChanged
 
+-- | Takes a function that manipulates the Dhall AST of the Config, and tries to run it
+--   on the current config. If it succeeds, it writes back to file the result returned.
+--   Note: it will pass in the parsed AST, not the resolved one (so e.g. imports will
+--   still be in the tree). If you need the resolved one, use `ensureConfig`.
+withConfigAST'
+  :: (HasLogFunc env, HasConfigPath env)
+  => (Expr -> RIO env Expr) -> RIO env Bool
+withConfigAST' transform = do
+  ConfigPath path <- view (the @ConfigPath)
+  rawConfig <- liftIO $ Dhall.readRawExpr path
+  case rawConfig of
+    Nothing -> die [ display $ Messages.cannotFindConfig path ]
+    Just (header, expr) -> do
+      newExpr <- transform $ Dhall.Core.denote expr
```

Comment:
This duplicates `withConfigAST` but doesn't use the `transformMOf Dhall.subExpressions``